Update master (current default branch)

i3anaan opened the pull request.

blacksph3re committed:
Added traefik support and played around with a go container

blacksph3re committed:
Merged mongoui PR

blacksph3re committed:
Added service-registry

blacksph3re committed:
The service-registry is now creating an additional access token

blacksph3re committed:
Added oms-applications

blacksph3re committed:
Registry now made the core obsolete

blacksph3re committed:
Updated the registry and applications to a first working version

blacksph3re committed:
Added frontend git

blacksph3re committed:
Changed format of getModules.json

linuxbandit committed:
Add repos of submodules

You forgot to add them :)

linuxbandit committed:
Make it work for vagrant machine as well (READ ME)

WARNING: This change introduces a modification of the localhost name used to access the app. In order to make it work into a pure docker environment, add into /etc/hosts an alias to localhost as "appserver"

linuxbandit committed:
Add a utility makefile & set provisioner to always run

invoke it with make <target name>, see the vagrant bootstrapper for details

the last provisioner now will always run (i.e. on every starting of the VM the app goes up as well)

README was updated to reflect usage with makefile and usage of Traefik

linuxbandit committed:
Edit .editorconfig to account makefile

linuxbandit committed:
Add other utilities to Makefile

linuxbandit committed:
Start "docker way" work

Add "docker way" for some dockerfiles and  the compose file (newer version)

linuxbandit committed:
Add docker way for omscore

linuxbandit committed:
Add docker way for events

linuxbandit committed:
Add docker way for service registry

linuxbandit committed:
Add docker way for applications

linuxbandit committed:
Add network setup, bump to version 3

Now docker swarm can accept the same file, with some annotations though

linuxbandit committed:
Refactor to have stuff near their logical belonging

Backends together, similar containers together, logs, networks

linuxbandit committed:
Add docker-stack file for swarm (it is actually a symlink :p )

linuxbandit committed:
Merge pull request #1 from linuxbandit/feat-swarm_ready

Feat: swarm ready

linuxbandit committed:
Merge pull request #16 from linuxbandit/traefik

Improve "Add Traefik as API gateway/reverse proxy"

linuxbandit committed:
Make it just start

Decoupled docker-compose and docker-stack (former is v2 and latter v3)
Dockerfile was corrected
Makefile was removed a command just for safety in case of non-vagrant
use

linuxbandit committed:
Merge pull request #17 from linuxbandit/traefik

Make it just start

blacksph3re committed:
Added a hopefully working dev version without the need to build images, relying on docker hub

blacksph3re committed:
Added gulp watch to omscore

blacksph3re committed:
Fixed broken registration process except for missing core functionality

blacksph3re committed:
Changed traefik config to accept all domain names

blacksph3re committed:
Fixed core bootstrapping

blacksph3re committed:
Fixed core bootstrapping, again

blacksph3re committed:
Updated git submodules

blacksph3re committed:
Changed the bootstrap procedure to run as laradock user

blacksph3re committed:
Another bug in the bootstrapping process upon empty install

blacksph3re committed:
Updated serviceregistry

blacksph3re committed:
Added staging environment with cloned repos, not working yet

blacksph3re committed:
Core is working, events and application both return a 404 though

blacksph3re committed:
Added working staging environment

blacksph3re committed:
Renamed docker folder to docker-prod

i3anaan committed:
Add IP address support to traeffik

Required to run on Fraktis

i3anaan committed:
Added php-intl, required by the core

i3anaan committed:
Removed docker-prod

i3anaan committed:
Updated oms-core submodule

i3anaan committed:
Removed unused submodules.

i3anaan committed:
Removed oms-serviceregistry folder

i3anaan committed:
Disabled some microservices.

i3anaan committed:
Corrected removing wrong core.

i3anaan committed:
Readded omscore-data.

i3anaan committed:
Traefik log level debug > info

i3anaan committed:
Removed unused container dockerfiles.

i3anaan committed:
Made single flexible build/tag/push -> setup script.

i3anaan committed:
Added usage example.

i3anaan committed:
Fixed some Building -> Pushing occurences.

i3anaan committed:
Renamed core .env to example.env

i3anaan committed:
Added echo explaining API Key output.

i3anaan committed:
Improved script runtime feedback.

i3anaan committed:
Spotted missing space.

i3anaan committed:
Moved omscore docker files to oms-core repository

i3anaan committed:
Updated oms-core submodule

i3anaan committed:
Updated core commit.

i3anaan committed:
Updated core commit once again.

i3anaan committed:
Moved oms-events related Dockerfiles in their repos.

i3anaan committed:
Fixed 2 rebel 'p's

i3anaan committed:
Removed staging option.

i3anaan committed:
Renamed folder docker-dev > docker

i3anaan committed:
Added simple oms-docker test

i3anaan committed:
Made Travis build instead of build the containers.

i3anaan committed:
small formatting correction/

i3anaan committed:
Updated readme slightly.

i3anaan committed:
Merge pull request #22 from AEGEE/stabilize-derk

Stabilized docker installation

i3anaan committed:
Bumped oms-core version. (after-merge cleanup)

i3anaan committed:
Bumped core version

blacksph3re commented:
I am strongly against merging this PR because it would make it really difficult for me to work on oms applications. It's true that I didn't work on it the recent days but still I hope to be able to take it up again. Also, why do you consider the serviceregistry as unused? My thoughts on the staging version I have expressed in the #operations channel and thus I would also be sad if my effort in this is disregarded, though I know that it is not yet a production ready setup (the volume mounts in the docker-compose.yml would have to be adjusted)

i3anaan commented:
@blacksph3re First off, please note that this version is the latest on the dev branch, this was merged ~4 days ago after me, Fabrizio and Sergey agreed to https://github.com/AEGEE/oms-docker/pull/22

To address some of your concerns:

> Also, why do you consider the serviceregistry as unused?

I removed it and the system still worked, did not notice any degraded functionality.

> My thoughts on the staging version I have expressed in the #operations channel

Taking from my comment on https://github.com/AEGEE/oms-docker/pull/22:

```
Removed differentiation between dev/stage/prod. (containers still have tags).
The reason for this was that the staging files we had were very similar to the dev files we had, causing confusion and danger of forgetting to maintain both files.
In the end, I (and @linuxbandit (who is here)) decided that the need for clarity and easy to maintain code was higher than that of having separate deployment modes.
In the future, we might end up adding it back again, but for now, it is overkill.
```
> and thus I would also be sad if my effort in this is disregarded
```
PS: If anyone can implement this without the need for having extra files, I would be very interested in this! (using an extra parameter + if statements or something along those lines.)
```

> though I know that it is not yet a production ready setup (the volume mounts in the docker-compose.yml would have to be adjusted)

**It was not just not production ready, it was also not development ready. We needed stabilization, we needed moving dockerfiles to the services' repositories and we needed a check if the integration of the entire system worked.**


Some extra comment:
Please do not regard this as final and perfect, but for the state of development that we are in, this was necessary. I still very much welcome some of the changes you proposed, the difference between dev/stage/prod is cool, yet more of a future thing. Additionally, I know there are some things wrong regarding usage of docker, but once again: This was better than what we had. Regarding pulling in Master: I figured why not? This is probably the most stable it will be in a while.

i3anaan committed:
Fixed omscore-nginx bug, added --reset command to oms-setup.sh

i3anaan committed:
Integration tests check if containers are running.

i3anaan committed:
Fixed traefik not knowing any commands > run detached.

i3anaan committed:
Added deploy.sh file, bumped core.

i3anaan committed:
Updated deploy.sh

i3anaan committed:
Moved logs to show after sleep.

i3anaan committed:
Bumped core version.

i3anaan committed:
Added instructions on where to place deploy.sh

linuxbandit commented:
Only thing I do not understand is looking at the git history: my changes are there and also nico's after mine (i.e. `traefik` is there and `stabilise` too - but they were 2 parallel ones!)

Other than that it is good as discussed

i3anaan committed:
oms_setup.sh now exits on the first encountered error (apart from --reset)

i3anaan committed:
Improved testing: Added local loopback test.

i3anaan committed:
README and LICENSE (#24)

* Redid readme.md

* Fixed folder name.

* Added license.

* Added title.

* Added missing space.

* Bumped core.

* Updated README slightly.

* Added links to the intro sentence.

* Corrected OManagementS > OMemberS.

linuxbandit commented on README.md:
```diff
@@ -1,32 +1,39 @@
-
-Open you browser on `http://appserver` and login with the default credentials. Navigate to the modules tab and activate all the microservices
-
+*This is assuming you are running a Linux installation, for other operating systems virtualization is recommended.*
```
Where is it written about the virtualisation (how to virtualise)?

i3anaan replied:
The more detailed installation instructions contain a link to an article about it:
https://oms-project.atlassian.net/wiki/spaces/GENERAL/pages/17235970/Installation

If you still want to do vagrant stuff feel free, but to be honest I felt like this was the simpler solution: giving the User more freedom in terms of how to get Ubuntu running, be at native, VirtualBox, Vagrant, or any other way. Besides, setting up VirtualBox is not really hard and should be well within the capabilities of someone supposed to work on the project.

linuxbandit replied:
Yes I wanted to ask if it was there (I am limited on a small screen so bear with me for not checking)

The vagrant is more for a 'one-liner' that makes it set up, that's why I asked

i3anaan committed:
Core env files update. (#26)

* Core env files update.

* Added check for the default admin password for core.

Bumped core.

i3anaan committed:
Corrected TAG > RESET in oms_setup.sh (#27)

linuxbandit reviewed (approved):
Tag it :p 

0.4.0 ?

i3anaan committed:
Merge pull request #25 from AEGEE/PROV-18

oms_setup.sh now exits on the first encountered error (apart from --reset)

i3anaan committed:
Merge branch 'dev' of https://github.com/AEGEE/oms-docker into testing

i3anaan commented:
NOTE: Default branch has been changed to dev.

i3anaan committed:
Merge pull request #28 from AEGEE/testing

Improved testing: Added local loopback test.

i3anaan committed:
Split OMS-core and traefik.

i3anaan committed:
Added run.sh file, improved Travis tests.

i3anaan committed:
Disabled cache for building, removed container selection.

i3anaan committed:
Removed double setup for travis LOL

i3anaan committed:
Improved oms.sh (prev. run.sh)
Now uses a list of docker-compose folders.
Support for pushing containers
Deprecated and removed docker/oms_setup.sh

i3anaan committed:
Updated deploy script.

i3anaan committed:
Fixed travis test using old script name.

i3anaan committed:
Doubled given setup time during tests (60 -> 120)

i3anaan committed:
Reworked oms.sh to work with -f option

Also put the global/common OMS services into a oms-global folder
Added a docker-compose environment file

i3anaan committed:
Upgraded docker-compose to version 3.3

Removes the omscore_bootstrap container

i3anaan committed:
Downgraded docker-compose to version 3.2

And fixed Travis tests

i3anaan committed:
Updated file location for LocalLoopbackTest.sh

i3anaan committed:
Paths are now relative to script location instead of execution location

i3anaan committed:
Merge pull request #31 from AEGEE/docker-upg

Docker split

i3anaan commented:
Let's close this until we have `oms-events` and the `registry` in `dev` again
